In [13]:
import numpy as np
import pandas as pd

In [8]:
df=pd.read_excel('Sales_Data.xlsx')

In [12]:
df.head().T

0                             1  \
ORDERNUMBER                           10424                         10425   
QUANTITYORDERED                          50                            38   
PRICEEACH                            240.02                        155.13   
ORDERLINENUMBER                           6                            12   
SALES                               12001.0                       5894.94   
ORDERDATE               2020-05-31 00:00:00           2020-05-31 00:00:00   
DAYS_SINCE_LASTORDER                     50                           150   
STATUS                           In Process                    In Process   
PRODUCTLINE                    Classic Cars                  Classic Cars   
MSRP                                    214                           147   
PRODUCTCODE                        S10_1949                      S10_4962   
CUSTOMERNAME          Euro Shopping Channel             La Rochelle Gifts   
PHONE                        (91) 555 94 44                    40.67.8555   
ADDRESSLINE1             C/ Moralzarzal, 86  67, rue des Cinquante Otages   
CITY                                 Madrid                        Nantes   
POSTALCODE                            28034                         44000   
COUNTRY                               Spain                        France   
CONTACTLASTNAME                      Freyre                       Labrune   
CONTACTFIRSTNAME                      Diego                        Janine   
DEALSIZE                              Large                        Medium   

                                          2                             3  \
ORDERNUMBER                           10424                         10425   
QUANTITYORDERED                          49                            33   
PRICEEACH                            162.64                         142.2   
ORDERLINENUMBER                           3                             4   
SALES                               7969.36                        4692.6   
ORDERDATE               2020-05-31 00:00:00           2020-05-31 00:00:00   
DAYS_SINCE_LASTORDER                    226                           376   
STATUS                           In Process                    In Process   
PRODUCTLINE                Trucks and Buses              Trucks and Buses   
MSRP                                    136                           118   
PRODUCTCODE                        S12_1666                      S12_4473   
CUSTOMERNAME          Euro Shopping Channel             La Rochelle Gifts   
PHONE                        (91) 555 94 44                    40.67.8555   
ADDRESSLINE1             C/ Moralzarzal, 86  67, rue des Cinquante Otages   
CITY                                 Madrid                        Nantes   
POSTALCODE                            28034                         44000   
COUNTRY                               Spain                        France   
CONTACTLASTNAME                      Freyre                       Labrune   
CONTACTFIRSTNAME                      Diego                        Janine   
DEALSIZE                              Large                        Medium   

                                          4  
ORDERNUMBER                           10424  
QUANTITYORDERED                          54  
PRICEEACH                             133.0  
ORDERLINENUMBER                           5  
SALES                                7182.0  
ORDERDATE               2020-05-31 00:00:00  
DAYS_SINCE_LASTORDER                    427  
STATUS                           In Process  
PRODUCTLINE                Trucks and Buses  
MSRP                                    116  
PRODUCTCODE                        S18_1097  
CUSTOMERNAME          Euro Shopping Channel  
PHONE                        (91) 555 94 44  
ADDRESSLINE1             C/ Moralzarzal, 86  
CITY                                 Madrid  
POSTALCODE                            28034  
COUNTRY                 

In [5]:
df.shape

(2747, 20)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2747 entries, 0 to 2746
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   ORDERNUMBER           2747 non-null   int64         
 1   QUANTITYORDERED       2747 non-null   int64         
 2   PRICEEACH             2747 non-null   float64       
 3   ORDERLINENUMBER       2747 non-null   int64         
 4   SALES                 2747 non-null   float64       
 5   ORDERDATE             2747 non-null   datetime64[ns]
 6   DAYS_SINCE_LASTORDER  2747 non-null   int64         
 7   STATUS                2747 non-null   object        
 8   PRODUCTLINE           2747 non-null   object        
 9   MSRP                  2747 non-null   int64         
 10  PRODUCTCODE           2747 non-null   object        
 11  CUSTOMERNAME          2747 non-null   object        
 12  PHONE                 2747 non-null   object        
 13  ADDRESSLINE1      

In [10]:
df.describe()

ORDERNUMBER  QUANTITYORDERED    PRICEEACH  ORDERLINENUMBER  \
count   2747.000000      2747.000000  2747.000000      2747.000000   
mean   10259.761558        35.103021   101.098951         6.491081   
std       91.877521         9.762135    42.042548         4.230544   
min    10100.000000         6.000000    26.880000         1.000000   
25%    10181.000000        27.000000    68.745000         3.000000   
50%    10264.000000        35.000000    95.550000         6.000000   
75%    10334.500000        43.000000   127.100000         9.000000   
max    10425.000000        97.000000   252.870000        18.000000   

              SALES  DAYS_SINCE_LASTORDER         MSRP  
count   2747.000000           2747.000000  2747.000000  
mean    3553.047583           1757.085912   100.691664  
std     1838.953901            819.280576    40.114802  
min      482.130000             42.000000    33.000000  
25%     2204.350000           1077.000000    68.000000  
50%     3184.800000           1761.000000    99.000000  
75%     4503.095000           2436.500000   124.000000  
max    14082.800000           3562.000000   214.000000

* There are no null values in the dataset.
* orderdate is the only datetime variable
* there are 2747 rows of data.

* From september to November the sales get very high and as the year ends the sales decreases.
* Overall trend of the sales is going upwards.
* For large dealsize March to May is showing an increase in sales.
* USA is the largest consumer.
* As sales increases the quantity of order also increases.
* Madrid records the highest sales among cities.
* Australia, Canada, Finland, France, Germany, Italy, Norway, UK and USA has wider outreach and high sales among the countries.
* For large deals the average price of the products are also high and for small it is less.
* The average sales of large deals is 8200 , medium 4400 and for small it is 2000.
* Medium deals has the highest amount of quantity ordered about 52000, small it is 38000 and for large it is 7200.
* Medium deals generate 60% of sales while small produces 27% and large produces 13%.
* Euro shopping channel, Mini gifts distributors, Austrailian collectors, Muscle machine and La Rochelle gifts are the top 5 consumers in terms of sales generated.
* Royal belge, Microscale INC, Auto-moto classics INC, Atelier graphique, Boards and Toys CO are the bottom 5 customers in terms of sales generated.
* Classic Cars generate the maximum amount of sales about 40% and Trains contribute the lowest about 2.3%.

#### Top Countries having highest sales among different productline:

##### Classic cars: USA, Spain, France, Austrailia and UK 
##### Motorcycles: USA, France, Austrailia and Spain
##### Planes: USA, France, Austrailia, Spain and Italy
##### Ships: USA, Spain, UK and France
##### Trains: USA, Spain and France
##### Trucks and Buses: USA, Spain, France, Singapore and Austrailia
##### Vintage cars: USA, Spain, UK, France and Italy


* For motor cycles, planes, ships, Trains and vintage cars Small deals are higher in amount.
* Except classic cars the average price of each productline is higher than the average price of its MSRP.
* Classic cars and the Trucks and Buses has the highest average price for its parts.

In [14]:
df=pd.read_csv('dataset_group.csv')

In [15]:
df.head()

Date  Order_id        Product
0  01-01-2018         1         yogurt
1  01-01-2018         1           pork
2  01-01-2018         1  sandwich bags
3  01-01-2018         1     lunch meat
4  01-01-2018         1   all- purpose